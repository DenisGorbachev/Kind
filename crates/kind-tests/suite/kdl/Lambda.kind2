#keep
CoolFn : Data.U60 -> Data.U60 {
  (x: Data.U60) => (* 2 x)
}

#keep
CoolFnApp (n: Data.U60) : Data.U60 {
  let lam = (x: Data.U60) => ((CoolFn) x)
  (lam n)
}