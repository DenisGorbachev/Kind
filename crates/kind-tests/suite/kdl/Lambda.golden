fun (CoolFnApp n) {
  (CoolFnApp x0) = (!@x0.0 (!@x1 (!@x1.0 (!x1.0 x0.0) x1) @x2 (!@x2.0 (!(CoolFn) x2.0) x2)) x0)
}

fun (CoolFn ) {
  (CoolFn) = @x0 (!@x0.0 (& (* #2 x0.0) #1152921504606846975) x0)
}

